crates/router/src/core/payments/flows/authorize_flow.rs

```diff
@@ -155,36 +167,24 @@ impl types::PaymentsAuthorizeRouterData {
                 );
 
                 logger::debug!(completed_pre_tasks=?true);
+
                 if self.should_proceed_with_authorize() {
                     self.decide_authentication_type();
                     logger::debug!(auth_type=?self.auth_type);
-                    let resp = services::execute_connector_processing_step(
```

Comment:
this block is a conditional block for `self.should_proceed_with_authorize()` condition, in decide flow the same is established?

Reply:
Yes. I will refactor the method in another PR to decide to call connector or not based only on whether the connector request is passed. I did not take that up in this PR as it would be a big refactor and add risk